akka-javasdk/src/main/java/akka/javasdk/impl/agent/CoreMemoryClient.java

```diff
@@ -9,40 +9,74 @@
 
+import java.util.Optional;
+
 /**
  * INTERNAL USE
  * Not for user extension or instantiation
  */
 @InternalApi
 public final class CoreMemoryClient implements CoreMemory {
 
+  public record MemorySettings(
+      Boolean read,
+      Boolean write,
+      Optional<Integer> historyLimit
+  ) {
+    static MemorySettings disabled() {
+      return new MemorySettings(false, false, Optional.empty());
+    }
+
+    static MemorySettings enabled() {
+      return new MemorySettings(true, true, Optional.empty());
+    }
+  }
+
   private final Logger logger = LoggerFactory.getLogger(CoreMemoryClient.class);
   private final ComponentClient componentClient;
+  private final MemorySettings memorySettings;
+
+  public CoreMemoryClient(ComponentClient componentClient, Config memoryConfig) {
+    this.componentClient = componentClient;
+    this.memorySettings = memoryConfig.getBoolean("enabled") ? MemorySettings.enabled() : MemorySettings.disabled();
+  }
 
-  public CoreMemoryClient(ComponentClient componentClient) {
+  public CoreMemoryClient(ComponentClient componentClient, MemorySettings memorySettings) {
     this.componentClient = componentClient;
+    this.memorySettings = memorySettings;
   }
 
   public void addInteraction(String sessionId,
                              String componentId,
                              ConversationMessage.UserMessage userMessage,
                              ConversationMessage.AiMessage aiMessage) {
-    logger.debug("Adding interaction to session: {}", sessionId);
-    componentClient.forEventSourcedEntity(sessionId)
-        .method(ConversationMemory::addInteraction)
-        .invoke(new ConversationMemory.AddInteractionCmd(componentId, userMessage, aiMessage));
+    if (memorySettings.write()) {
+      logger.debug("Adding interaction to session: {}", sessionId);
+      componentClient.forEventSourcedEntity(sessionId)
+          .method(ConversationMemory::addInteraction)
+          .invoke(new ConversationMemory.AddInteractionCmd(componentId, userMessage, aiMessage));
+    } else {
+      logger.debug("Memory writing is disabled, interaction not added to session: {}", sessionId);
+    }
   }
 
   @Override
   public ConversationHistory getHistory(String sessionId) {
-    var history = componentClient.forEventSourcedEntity(sessionId)
-        .method(ConversationMemory::getHistory)
-        .invoke();
-    logger.debug("History retrieved for sessionId={} size={}", sessionId, history.messages().size());
-    return history;
+    if (memorySettings.read()) {
+      var history = componentClient.forEventSourcedEntity(sessionId)
+          .method(ConversationMemory::getHistory)
+          .invoke(new ConversationMemory.GetHistoryCmd(memorySettings.historyLimit));
+      logger.debug("History retrieved for sessionId={} size={}", sessionId, history.messages().size());
+      return history;
+    } else {
+      logger.debug("Memory reading is disabled, history not retrieved for sessionId: {}", sessionId);
```

Comment:
many different variations of how the log parameters are defined
`sessionId: {}`
`sessionId={}`

Mostly, we try to use `[]` around the parameters
`sessionId [{}]`

Reply:
tks, using `sessionId [{}]` in all places